app/README.md

```diff
@@ -52,9 +52,127 @@ committing changes, otherwise the tests will be skipped on CI.
+can be composed to create rich user experiences.
+
+Most of the components defined by the design system can be conveniently seen on
+one page within the app UI by pressing <kbd>shift</kbd> + <kbd>D</kbd>. This
+will bring up the Design System Guide modal. To close the modal, hit
+<kbd>ESC</kbd>.
+
+The purpose of the Guide is to be able to see how changes to components affect
+all variations in one glance. Newly created components should be added to the
+Guide, explaining all the different variations and use cases a component can be
+used.
+
+
+#### File structure
+
+To be able to find code and make changes quickly we organize our components in
+groups defined by use-case, re-usability and complexity. These groups are:
+
+  - `DesignSystem/Primitives`: simple, yet highly reusable components like
+    typography, buttons, form elements, spacing, positioning and other
+    utilities.
+
+    Components in this group are usually just wrappers around standard HTML
+    elements with custom styling.
+
+    There are currently two ways of organizing primitives:
+
+      - all-in-one components where the type of the component is passed down
+        via a `variant` prop. This is for components which have a very
+        similar markup, but whose styling differs across variants.
+        Examples in this category are: buttons, typography and positioning
+        helpers.
+
+      - namespaced components, where the markup is very different across
+        variants, like: form elements and icons.
+
+    To figure which way to write a new primitive component, start by looking at
+    how it's going to be used in the code and then go for the better
+    ergonomics.
+
+    All public primitives are exported via a central `index.js` file, which
+    makes usage as easy as:
+
+    ```
+    <script>
+      import { Button, Text, Icon, Input } from "../DesignSystem/Primitives";
+    </script>
+
+    <Icon.Home />
+    <Button variant="secondary">OK</Button>
+    <Text variant="hugeTitle">Radicle</Text>
+    ```
+
+
+  - `DesignSystem/Components`: reusable low-to-high complexity components,
+    layouts and single-use fragments which are extracted from screens or other
+    components for the sake of code readability.
+
+    Sub-folders in this category should only be created for breaking up larger
+    components into smaller fragments. If a component is broken up in
+    fragments, make sure to only export the component for public use, not the
+    fragments.
+
+    Single-use components that are extracted from screens and used by only one
+    screen should be prefixed with the name of the screen, for example:
+    `DesignSystemGuideSection.svelte`.
+
+
+There is a third group of components, which is not part of the Design System:
+"Screens". Screens bring together components from the Design System forming
+what a user sees in the UI as a whole screen. More complex UX flows can be
+broken down into multiple sub-screens, in this case a top-level screen should
```

Comment:
If I understand this correctly sub-screens are full screens, just switche based on the state of the parent screen. If that assumption is correct, there might be a possibility to distinguish these even further. To avoid the subtleties between screen and sub-screen. For example introducing the concept of page or view which holds 1..N screens.

Reply:
It doesn't have to be a full screen necessarily. It could also be a tab, containing just a part of the top-level screen (like in project registration).

I could change the wording a bit if sub-screens are confusing, something like:

```
`Screens` bring together components from the Design System forming what a user
in the UI sees as a whole screen. More complex screens can be broken down into
multiple partials, in this case a screen should contain common data fetching and
routing for the partials. Partials should be placed in a directory named after the screen.
```

Maybe this way we can also avoid having single-use components in the `DesignSystem/Components` directory.

Reply:
Alternatively we could call them fragments, similar to how we do with broken-up components.

Reply:
That helps a lot already, especially the part about state and routing as I think this is the most important outcome - to determine for which entity to draw the line in regards for example expecting to be parameterised via `location`. I have no strong opinions on the names themselves as long as the concepts/entities are distinct and rules have been established what is allowed for either. i.e.: `A Screen holds a single store which is used to transport down information to its fragments...`